sql/statesql/migrations/state_0021_migration.go

```diff
@@ -61,26 +62,42 @@ func (m *migration0021) Apply(db sql.Executor, logger *zap.Logger) error {
 		progress := float64(processed) * 100.0 / float64(total)
 		logger.Info("processed ATXs", zap.Float64("progress [%]", progress))
 		if processed >= total {
-			return nil
+			break
 		}
 	}
+
+	if err := m.applyIndices(db); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 func (m *migration0021) applySql(db sql.Executor) error {
-	query := `CREATE TABLE posts (
-		atxid  CHAR(32) NOT NULL,
-		pubkey CHAR(32) NOT NULL,
-		prev_atxid  CHAR(32),
-		prev_atx_index INT,
-		units  INT NOT NULL,
-		UNIQUE (atxid, pubkey)
-	);`
+	query := dedent.Dedent(`
+		CREATE TABLE posts (
+		    atxid CHAR(32) NOT NULL,
+		    pubkey CHAR(32) NOT NULL,
+		    prev_atxid CHAR(32),
+		    prev_atx_index INT,
+		    units INT NOT NULL
+		);`)
```

Comment:
I don't get why we need a dependency for this? This could also be done by this:

```suggestion
	query := "CREATE TABLE posts (\n" + 
		"\t\tatxid CHAR(32) NOT NULL,\n" +
		"\t\tpubkey CHAR(32) NOT NULL,\n" +
		"\t\tprev_atxid CHAR(32),\n" +
		"\t\tprev_atx_index INT,\n" +
		"\t\tunits INT NOT NULL\n" +
		"\t);\n"
```

It doesn't solve the issue that if the *.sql file doesn't match the coded migration exactly (including whitespace and possibly other meaningless differences) the check for unchanged schema will fail.

Reply:
The idea of using dedent is to improve readability of `schema.sql` and `.schema` command output. The schema drift detection mechanism is not affected. If this needs to be done of the price of having to write less readable escaped string literals in Go code, than it basically doesn't make too much sense

Reply:
I don't understand... the change to `Diff` already solves the whitespace difference issue. So even without changing anything to this query the schema comparison would pass.

`dedent` to me solves a problem that doesn't exist - it removes whitespace from a string that we could ensure ourselves isn't there and also doesn't matter any more in the comparison?